docs: add our philosophy (#141)
open-wc/open-wc: docs/guide/README.md

- [Publishing](/publishing/)
- [Automating](/automating/)

## Why `open-wc`

We want to provide the community with well-known and experience-tested recommendations for their web component projects. Everything we do here is to encourage and support development with these open standards.

### Our Philosophy

The goal of Open Web Components is to empower developers with powerful and battle-tested tools for creating and sharing open source web components.

Many web developers have experienced the dreaded "Javascript Fatigue". With our recommendations, we hope you'll enjoy the peace of mind that comes from having a well-known default solution for almost everything. From IDE to CI, `open-wc` has got you covered.

> We want web component development to be accessible and approachable for people, so our recommendations aim to:

Author: ```suggestion
We want web component development to be accessible and approachable for everyone, regardless of your background, so our recommendations aim to:
```

- Be easy to use
- Be ready to use
- Give you that 'it just works' feeling

### Why Web Components

We believe that the open web is an ideal platform for application development. It's free, open, and standardized nature reenforces user's privacy and control while enabling developers to produce useful apps.

#### Historical Background

> **tl;dr:** Libraries provided the inspiration to standardize the web's component model, which became a practical reality in 2019 after many years of work.

Author: ```suggestion```

Reviewer: The motivation for putting this here was because the history section is a little long (although I did revise a few times to make it as concise as possible)

Author: I think the history section is short enough to not have to include a tl;dr, I also feel like a tl;dr is more appropriate in e.g. a blog post instead of here

Reviewer: what do you think of splitting this page into `our philosophy` and `why web components`?

Reviewer 2: I think there will be more sections => I'm still missing our rationals for what we recommend (or our requirements that need to be met before we recommend?)

The World Wide Web was originally just text documents connected by hyperlinks. As technologies like JavaScript and techniques like <abbr title="Asynchronous JavaScript and XML">AJAX</abbr> and <abbr title="Representational State Transfer">REST</abbr> entered the picture, the web slowly developed into a platform for full-fledged application development.

> At first, JavaScript libraries like jQuery were absolutely required. jQuery gave developers a common set of tools across browsers with extremely different feature sets. Eventually jQuery features became standard browser features. The pattern of libraries innovating then browsers standardizing continued. As the "[browser wars](https://www.wikiwand.com/en/Browser_wars)" died down, the pace of standardization increased, and many new features came to the web.

Author: ```suggestion
At first, JavaScript libraries like jQuery were absolutely required. jQuery gave developers a common set of tools across browsers with extremely different feature sets. Eventually jQuery features became standard browser features. The pattern of libraries innovating, and browsers consequently standardizing continued. As the "[browser wars](https://www.wikiwand.com/en/Browser_wars)" died down, the pace of standardization increased, and many new features came to the web.
```

> In the last several years, a component-based model for web app development became popular, and the JavaScript community blossomed with a wide variety of libraries and approaches. Work on standardizing the web's component model began at Google in 2012. After several years of open development, the web components standards became implemented across all major browsers in 2019.

Author: ```suggestion
In the last several years, a component-based model for web application development was popularized, and the JavaScript community blossomed with a wide variety of libraries and approaches. Work on standardizing the web’s native, standards-based component model began at Google in 2012, and after several years of open development, they were successfully implemented across all major browsers in 2019.
```

#### Technical Benefits

> Working with built-in browser features has several advantages over purely JavaScript-based frameworks.

Author: ```suggestion
Working with built-in browser features has several advantages for us to make use of.
```

##### Performance
Just like `querySelector` reduces developers' dependence on jQuery, the web components standards reduces the size and complexity of component libraries and frameworks. This ultimately benefits users by decreasing page load times and increasing app performance.

##### Future Proofing

> Using standard components protects your app from framework churn. The web has an extremely strong tradition of backwards-compatibility. Standards bodies have consistently gone out of their way to maintain legacy APIs.

Author: ```suggestion
Using the browser's standards-based component model will increase the longevity of your application. The web has an extremely strong tradition of backwards-compatibility. Standards bodies have consistently gone out of their way to maintain legacy APIs.
```

Reviewer: I want to keep the bit about framework churn in there. I don't think it's throwing shade because we're not pointing any specific fingers. Do you think it's dishonest (because what's to say that a wc lib won't churn)?

Author: I agree that it's not pointing specific fingers at any specific framework, but to seems like it still points a finger a frameworks in general, where it could be avoided. 

Instead of saying something that could be considered negative about frameworks and possibly elicit reactions ("yeah, _but_ ...!""), we can convey the same message by saying something positive about web components instead

I prefer to maintain a more neutral, inviting approach, especially to people coming from frameworks

Reviewer: I mean, if web components have this advantage over frameworks, it's not finger pointing to display that.

Author: The fact that it's true (in which I agree with you) does not mean it's necessarily the best/friendliest way to communicate about it. Maybe we should have someone else chime in? @daKmoR ?

Reviewer: I'd like to hear ideas on a more inclusive way of communicating "if you've been burned by library churn in the past, wc might be a wayforward"

Author: I think my point is rather that we can avoid talking about framework churn. It's almost guaranteed to elicit bad response from framework-people, where we can be more welcoming to that demographic instead.

We can be positive about web components without having to be negative about frameworks. I don't think this page is the place to fuel the frameworks vs web components discussion.

Reviewer: I'm happy to defer to @daKmoR 's better judgement

Reviewer 2: uh a nice conversions we have going on here.

To be honest, I didn't know the term "framework churn" before 🙈 but to me, it felt rather rude and we should be as welcoming as possible. That said we probably could mention the term but not as an argument and not without a link.

```suggestion
Using the browser's standards-based component model will increase the longevity of your application. The web has an extremely strong tradition of backward-compatibility. Standards bodies have consistently gone out of their way to maintain legacy APIs. Webcomponent are even seen as [the end of framework churn](https://blog.ionicframework.com/the-end-of-framework-churn/).
```

I'm pretty sure you guys can make something better out of `Webcomponent are even seen as [the end of framework churn](https://blog.ionicframework.com/the-end-of-framework-churn/).` 🙈 

also maybe someone know a better article?

Reviewer 2: so can we change this one line and then merge it?
it's already very useful 💪

Author: I say for now leave it out until we reach more of a consensur/better way to word it. We still need to iterate on this page anyway.
